core/src/main/java/com/scalar/db/common/CheckedDistributedStorageAdmin.java

```diff
@@ -0,0 +1,281 @@
+package com.scalar.db.common;
+
+import com.scalar.db.api.DistributedStorageAdmin;
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.io.DataType;
+import com.scalar.db.util.ScalarDbUtils;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+public class CheckedDistributedStorageAdmin implements DistributedStorageAdmin {
+
+  private final DistributedStorageAdmin admin;
+
+  /**
+   * Whether to check if the namespace exists or not. Set false when the storage does not support
+   * namespaces.
+   */
+  private final boolean checkNamespace;
+
+  public CheckedDistributedStorageAdmin(DistributedStorageAdmin admin) {
+    this(admin, true);
+  }
+
+  public CheckedDistributedStorageAdmin(DistributedStorageAdmin admin, boolean checkNamespace) {
+    this.admin = admin;
+    this.checkNamespace = checkNamespace;
+  }
+
+  @Override
+  public void createNamespace(String namespace, Map<String, String> options)
+      throws ExecutionException {
+    if (checkNamespace && namespaceExists(namespace)) {
+      throw new IllegalArgumentException("Namespace already exists: " + namespace);
+    }
+
+    try {
+      admin.createNamespace(namespace, options);
+    } catch (ExecutionException e) {
+      throw new ExecutionException("Creating the namespace failed: " + namespace, e);
+    }
+  }
+
+  @Override
+  public void createTable(
+      String namespace, String table, TableMetadata metadata, Map<String, String> options)
+      throws ExecutionException {
+    if (checkNamespace && !namespaceExists(namespace)) {
+      throw new IllegalArgumentException("Namespace does not exist: " + namespace);
+    }
+    if (tableExists(namespace, table)) {
+      throw new IllegalArgumentException(
+          "Table already exists: " + ScalarDbUtils.getFullTableName(namespace, table));
+    }
+
+    try {
+      admin.createTable(namespace, table, metadata, options);
+    } catch (ExecutionException e) {
+      throw new ExecutionException(
+          "Creating the table failed: " + ScalarDbUtils.getFullTableName(namespace, table), e);
+    }
+  }
+
+  @Override
+  public void dropTable(String namespace, String table) throws ExecutionException {
```

Comment:
These `xxxxTable()` and `xxxxIndex()` should check namespace as well as `createTable()`? I think either is fine as long as they're consistent.

Reply:
In the operations that assumes the the target table exists, we don't have the namespace check. In that sense, they should be consistent. And the table check contains the namespace check, so I think that should be okay.